### ISSUE TITLE
Certain permissions are essential

### ISSUE THREAD

**Issue by xavier-GitHub76** [apache/superset#27765]
### Bug description

Hello,

I find permission management tricky because there is little documentation on the subject.
with Superset v3.1.1 (docker), I started from a role with no permission to try to understand the impacts of permissions.

I notice that certain essential functions rely on permissions. Here is the detail :

With no permission, I navigate on pages

**HOME PAGE**
the simple access on welcome page causes errors
![image](https://github.com/apache/superset/assets/82046143/ae22f6d6-8abb-45e9-b672-d28e65a1a819)
![image](https://github.com/apache/superset/assets/82046143/85e092c8-3023-4225-9ca7-a2cedddd9b9a)

The following permissions are therefore obligatory : 
- "can read on Dashboard" to list dashbords on home page
- "can read on Chart" to list charts on home page
- "can recent activity on Log"

The Create buttons are presents although the user has no permission. Click on it --> Error
![image](https://github.com/apache/superset/assets/82046143/e617d022-e001-497a-a93c-055a19b9b562)

**PARAMETERS/PROFIL**
OK, the page is displayed but why  a permission "can profile on Superset" exists ? It's seems unnecessary

**PARAMETERS/INFO**
Error "Acces is denied"
![image](https://github.com/apache/superset/assets/82046143/ea6cf926-ebc8-44c7-9792-37479b7bdf34)

The following permission is therefore obligatory : 
- "userinfoedit on UserDBModelView"

**Proposed solution**
if permission is imposed on everyone then it should not exist : 
- "can read on Dashboard"
- "can read on Chart"
- "can recent activity on Log"
- "userinfoedit on UserDBModelView"
could be deleted

Create buttons should be displayed only if permission (value to specify) are present
Same for Parameters/Info, it's  should be displayed only if permission "userinfoedit on UserDBModelView" is present

if permission is unnecessary then it should not exist : 
- "can profile on Superset" could be deleted

Best regards

### How to reproduce the bug

Create a role without permission
Create a user with this role
With this user, go on : 
- welcome page
- > Error dashbaords et charts listing
- > Error recent logs
- > Error on Create buttons (dashboards and charts)
- profile page
- > Displayed without permission
- info page
- > Error Access denied

### Superset version

3.1.1

### Python version

3.9

### Node version

16

### Browser

Chrome

### Additional context

Docker

### Checklist

- [ ] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [ ] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [ ] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by goldjee** [comment]
Yep, the whole permission system feels very clunky and too raw to manage. If a permission is necessary to be granted for a user to actually use the system, it should not be listed.

Also, it's worth mentioning that there are 176 permissions assigned to vanilla Admin role with no proper documentation. If it's the whole set of permissions, I believe they should be grouped by functionality so superusers could easily acknowledge what is assigned to any given role. It could be visualized as a table, like so:

| Functionality | Access | Modify | Delete |
| ------------- | ------- | ------- | ------ |
| Dashboard     | [x]         | [ ]         | [ ]       |

I believe such a permission view could be a great relief to anyone who administers Superset or conducts security audit on the installation.

**Comment by xavier-GitHub76** [comment]
I 'm OK with you.
Today, list of permissions is very difficult to use when the number of permissions is high.
For example, a dropdown list with 170 items is not ergonomic. 

I think it could be more simple to use if following rules : 

1. If an action is enable for everyone, a permission should not exist (like 'can profile on Superset')
2. If an action is not enable for everyone, a permission should exist (like 'can write on Dashbaord') and the buttons linked to theses permissions should be only displayed if permssion is present on role. This would avoid the presence of unnecessary buttons. They will inevitably cause a warning or an error
3. Permissions should be grouped by category to have smaller and easier to understand lists. The categories could be the main elements managed on Superset : 

- Database
- Dataset
- Chart
- Dashboard
- SqlLab
- Other 

Each category could be displayed with tabs : 
![image](https://github.com/apache/superset/assets/82046143/58ee0fdd-9c58-4089-967f-91029a1a17bb)

Your table  will be very more pratical to use : 
![image](https://github.com/apache/superset/assets/82046143/fb66ebd8-4bb1-45f5-a946-8ab94a8eaa52)

The administrator could thus define a role with simple boxes to check.
This would avoid having to interpret an permission based on a technical name.
Personally, I don't understand permissions like "can get value on KV"

**Comment by rusackas** [comment]
Just wanted to point out that @mistercrunch has a SIP open here: 
https://github.com/orgs/apache/projects/170/views/1?filterQuery=permissions&pane=issue&itemId=60729049

**Comment by sfirke** [comment]
There are a lot of valid criticisms and suggestions above.  I think this is most likely to be tackled as part of a massive overhaul of permissions - the permissions system needs a complete revisiting beyond the modest suggestions here. The latest discussion is SIP-131: https://github.com/orgs/apache/projects/170/views/1?filterQuery=permissions&pane=issue&itemId=62189210&issue=apache%7Csuperset%7C28377 

I'm going to convert this into a discussion, as it's not a "bug" per se that could conceivably be tackled on its own and this issue closed.
